File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/event/PlayerAttackEvent.java
package top.likoslupus.cellulosesz.api.event;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;

import java.util.Objects;
import java.util.Optional;
import java.util.UUID;

public final class PlayerAttackEvent extends AbstractCancellableEvent {

    private final CellPlayer player;
    private final Optional<UUID> targetPlayer;
    private final String targetType;

    public PlayerAttackEvent(
            CellPlayer player,
            Optional<UUID> targetPlayer,
            String targetType
    ) {
        this.player = Objects.requireNonNull(player, "player");
        this.targetPlayer = Objects.requireNonNull(targetPlayer, "targetPlayer");
        this.targetType = Objects.requireNonNull(targetType, "targetType");
    }

    public CellPlayer player() {
        return player;
    }

    public Optional<UUID> targetPlayer() {
        return targetPlayer;
    }

    public String targetType() {
        return targetType;
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/event/PlayerAttackEvent.kt
package top.likoslupus.cellulosesz.api.event

import top.likoslupus.cellulosesz.api.platform.CellPlayer
import java.util.*

public class PlayerAttackEvent(
    private val player: CellPlayer,
    private val targetPlayer: UUID?,
    private val targetType: String
) : AbstractCancellableEvent() {

    public fun player(): CellPlayer = player

    public fun targetPlayer(): UUID? = targetPlayer

    public fun targetType(): String = targetType

}
